src/persist/src/indexed/mod.rs

```diff
@@ -315,11 +337,29 @@ impl<K: Data, V: Data, U: Buffer, L: Blob> Indexed<K, V, U, L> {
             (k1, v1, t1).cmp(&(k2, v2, t2))
         });
 
+        // Trace batches are required to be consolidated and only have
+        // one update per (k, v, ts).
+        let mut consolidated_updates = vec![];
+
+        for ((k, v, t), group) in &updates
```

Comment:
Looks like dd has a method for this one https://docs.rs/differential-dataflow/0.12.0/differential_dataflow/consolidation/fn.consolidate_updates.html

Reply:
I totally forgot about this method! I used it here and in the future batch consolidation case (reshaped the vector to `(t, (k, v))`, applied `consolidate_updates` and then reshaped it back to the shape that we want. Lmk what you think of that approach (I think its a little awkward to copy the vector again at the end but its worth it to be able to reuse this function and avoid having to write our own `consolidate` impl)